Core/Core/MobileCore.cs

```diff
@@ -213,10 +212,44 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
             throw new ServiceModuleInstanceNotFoundException(String.Format("No instance has been registered for interface {0}", serviceClass.Name));
         }
 
-        public ServiceConfiguration GetServiceConfiguration(String type)
+        /// <summary>
+        /// Returns array of ServiceConfiguration, filtered by type
+        /// </summary>
+        /// <param name="type">type field of the configuration</param>
+        /// <returns>array of ServiceConfiguration</returns>
+        public ServiceConfiguration[] GetServiceConfigurationByType(String type)
+        {
+            List<ServiceConfiguration> listOfConfigs = new List<ServiceConfiguration>();
+            foreach (var item in servicesConfig)
+            {
+                if (item.Value.Type == type)
+                {
+                    listOfConfigs.Add(item.Value);
+                }
+            }
+            ServiceConfiguration[] arrayOfConfigs = listOfConfigs.ToArray();
+            return arrayOfConfigs;
+        }
+
+        /// <summary>
+        /// Returns the first instance of a ServiceConfiguration based on the type key
```

Comment:
This is a rather picky observation but what does exactly define "first" here?. There is no order in `servicesConfig`, right? Is it the same as saying "first occurrence"?